host/tests/rmfAudio_L3_TestCases/rmfAudio_test02_primaryJitterTest.py

```diff
@@ -0,0 +1,98 @@
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
+# *
+# *
+# http://www.apache.org/licenses/LICENSE-2.0
+# *
+# * Unless required by applicable law or agreed to in writing, software
+# * distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import os
+import sys
+import time
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path, "../"))
+
+from rmfAudio_L3_TestCases.rmfAudioHelperClass import rmfAudioHelperClass
+
+class rmfAudio_test02_primaryJitterTest(rmfAudioHelperClass):
+    """
+    Test class to open capture Primary Audio Data and check if there is jitter detected.
+
+    This class uses the `rmfAudioClass` to run primary data capture,
+    downloading necessary test assets, setting up RMF Audio Capture
+    and performing verification of audio capture rate.
+    """
+    def __init__(self):
+        """
+        Initializes the test class with test name, setup configuration, and sessions for the device.
+
+        Args:
+            None
+        """
+        self.testName  = "test02_primaryJitterTest"
+        super().__init__(self.testName, '2')
+
+    def testFunction(self):
+        """
+        The main test function that verifies primary data capture with input audio.
+
+        This function:
+        - Starts Primary Data capture.
+        - Tracks bytes received for Audio data on primary interface.
+        - Monitors bytes received in regular intervals
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        self.log.testStart(self.testName, '2')
+        result = True
+
+        # 1 for primary data capture (default), 2 for auxiliary data capture.
+        capture_type = 1
+        jitter_test_duration = 120 # jitter test duration in seconds
+        threshold = 16384 # 16K
+        jitter_interval = 100000 # 100ms
+        settings_update = 0 # settings_update=0 (no updates to default settings), capture_format=1, sampling_rate=1, fifo_size=1, threshold=1
+        # For settings update, set settings_update to 1, followed by value for capture_format, Sampling Frequency, FIFO size, threshold. 
+        # Use -1 for settings parameter that you want to retain default value for.
+        # Sample shows 1 for settings update, 0(racFormat_e16BitStereo) for capture format, 5(racFreq_e48000) for sampling rate, 16384 for FIFO and -1 to retain default threshold value.
+        # sample : self.testrmfAudio.updateSettings(capture_type, 1, 0, 5, 16384, -1)
+        test_type = 1 # test_type : 1 for byte counting tests (default), 2 for data tracking tests where audio data is captured.

```

Comment:
Are we capturing data for test_type = 2 as well?

Reply:
There are two types of tests we do for audio capture module. 
1 - tests like jitter, where only bytes received count is sufficient to validate test. 
2 - data capture tests, where we need to capture exact audio data received and validate it against reference audio. 

So the test_type is 1 for jitter tests, 2 for data capture tests.